libraries/chain/include/eosio/chain/transaction.hpp

```diff
@@ -127,6 +127,24 @@ namespace eosio { namespace chain {
          return account_name();
       }
 
+      account_name resource_payer()const {
+         auto resource_payer = first_authorizer();
```

Comment:
It would be more efficient to not take the time to retrieve the first_authorizer if resource_payer is in the trx extension.

Reply:
Thanks for the suggestion, i changed that part